lib/ec2imgutils/ec2uploadimg.py

```diff
@@ -621,19 +630,27 @@ def _execute_ssh_command(self, command):
         return stdout.read().strip().decode('utf-8')
 
     # ---------------------------------------------------------------------
-    def _find_equivalent_device(self, device_id):
-        """Try and find a device that should be the same device in the
-           instance than the one we attached with a given device id."""
-        if self.aborted:
-            return
-        device_letter = device_id[-1]
-        expected_name = '/dev/xvd%s' % device_letter
-        if self._device_exists(expected_name):
-            return expected_name
+    def _find_device_name(self, device_size):
+        """Match an attached volume by size"""
+        lsblk_out = json.loads(self._execute_ssh_command('lsblk -a -J'))
+        for device in lsblk_out['blockdevices']:
```

Comment:
Should this be checked first that there is a list of blockdevices? Or is it guaranteed to be an empty list if none are returned?